xlators/features/locks/src/common.c

```diff
@@ -968,11 +968,7 @@ grant_blocked_locks(xlator_t *this, pl_inode_t *pl_inode)
     pl_local_t *local = NULL;
     INIT_LIST_HEAD(&granted_list);
 
-    pthread_mutex_lock(&pl_inode->mutex);
-    {
-        __grant_blocked_locks(this, pl_inode, &granted_list);
-    }
-    pthread_mutex_unlock(&pl_inode->mutex);
+    __grant_blocked_locks(this, pl_inode, &granted_list);
 
     list_for_each_entry_safe(lock, tmp, &granted_list, list)
```

Comment:
This loop cannot be executed inside a mutex. It can take a significant amount of time and will increase the chances of lock contention. It could even lead to deadlock if during the unwind a new request is sent and the same lock needs to be taken.